src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -137,40 +136,36 @@ private static void validateHost(String host, String URL) {
     if (!actualHostURL.equals(expectedHostURL)) {
       throw new InvalidBlobURLException(
           "Ingest source is not a valid blob url: '"
-              + URL
+              + url
               + "'. "
               + "The host is expected to take the following format: {storageAccountName}.blob.core.windows.net");
     }
   }
 
   private static void validateContainerAndBlobNames(
-      String containerName, String blobName, String URL) {
+      String containerName, String blobName, String url) {
+
+    String azureContainerRegex = "^[a-z0-9](?!.*--)[a-z0-9-]{1,61}[a-z0-9]$";
+
+    if (!containerName.matches(azureContainerRegex)) {
+      throw new InvalidBlobURLException(
```

Comment:
Is an `InvalidBlobURLException` still the right exception type, or should this be renamed?

Reply:
I was thinking it was still ok given it is part of the URL

Reply:
And we use the exception for things like the file name in the path too so I think we probably should keep it here too?

Reply:
Fine by me!  I wasn't sure if the container name was considered part of the blob URL, or a separate entity.